src/precompiles.rs

```diff
@@ -42,11 +56,156 @@ pub fn istanbul_precompiles(
+        // Input slice format:
+        //      sender (20 bytes) - Eth address of the address that burned his erc20 tokens
+        //      amount (U256 le bytes) - the amount that was burned
+        //      recipient_account_id (bytes) - the NEAR recipient account which will receive NEP-141 tokens
+
+        //TODO: add method in Aurora connector and call promise `get_near_account_for_evm_token(context.caller)`
+        let nep141address = context.caller.to_string();
+
+        let mut input = input;
+
+        let mut sender = [0u8; 20];
+        sender.copy_from_slice(&input[..20]);
+        input = &input[20..];
+
+        let amount = U256::from_little_endian(&input[..32]).as_u128();
+        input = &input[32..];
+
+        let receiver_account_id: AccountId = String::from_utf8(input.to_vec()).unwrap();
+
+        (
+            nep141address,
+            NEP41TransferCallArgs {
+                receiver_id: receiver_account_id,
+                amount,
+                memo: None,
+            },
+        )
+    };
+
+    let promise0 = sdk::promise_create(
+        nep141account,
+        b"ft_transfer",
+        BorshSerialize::try_to_vec(&args).ok().unwrap().as_slice(),
+        0,
+        GAS_FOR_FT_TRANSFER,
+    );
+
+    sdk::promise_return(promise0);
+}
+
+#[allow(dead_code)]
+fn exit_to_ethereum(input: &[u8], context: &Context) {
+    // TODO: Determine the correct amount of gas
+    const GAS_FOR_WITHDRAW: Gas = 50_000;
+    // if Self::required_gas(input)? > target_gas {
+    //     return Err(ExitError::OutOfGas);
+    // }
+
+    let (nep141account, serialized_args) = if context.apparent_value != U256::from(0) {
+        // ETH transfer
+        //
+        // Input slice format:
+        //      eth_recipient (20 bytes) - the address of recipient which will receive ETH on Ethereum
+
+        let eth_recipient: AccountId = String::from_utf8(input.to_vec()).unwrap();
+        let args = BridgedTokenWithdrawEthConnectorArgs {
+            amount: context.apparent_value.as_u128(),
+            recipient: eth_recipient,
+        };
+
+        (
+            String::from_utf8(sdk::current_account_id()).unwrap(),
+            BorshSerialize::try_to_vec(&args).ok().unwrap(),
+        )
+    } else {
+        // ERC-20 transfer
+        //
+        // This precompile branch is expected to be called from the ERC20 withdraw function
+        // (or burn function with some flag provided that this is expected to be withdrawn)
+        //
+        // Input slice format:
+        //      sender (20 bytes) - Eth address of the address that burned his erc20 tokens
+        //      amount (U256 le bytes) - the amount that was burned
+        //      eth_recipient (20 bytes) - the address of recipient which will receive ETH on Ethereum
+
+        //TODO: add method in Aurora connector and call promise `get_near_account_for_evm_token(context.caller)`
+        let nep141address = context.caller.to_string();
+
+        let mut input = input;
```

Comment:
Probably a good idea to make this a new variable like, `account_bytes`

Reply:
Also, naming conventions for `nep141address`: `nep141_address`.

Reply:
Done!